cpp/src/transform/transform.cpp

```diff
@@ -34,75 +34,173 @@ namespace cudf {
 namespace transformation {
 namespace jit {
 namespace {
-void unary_operation(mutable_column_view output,
-                     column_view input,
-                     std::string const& udf,
-                     data_type output_type,
-                     bool is_ptx,
-                     rmm::cuda_stream_view stream)
+
+using device_data_t = void*;
+
+std::vector<std::string> build_jit_typenames(mutable_column_view output,
+                                             std::vector<column_view> const& inputs)
+{
+  static constexpr auto SCALAR_STRIDE = 0;
+  static constexpr auto COLUMN_STRIDE = 1;
+
+  auto const column_type_name = [](data_type data_type, bool is_scalar) {
+    return jitify2::reflection::Template("cudf::transformation::jit::strided")
+      .instantiate(type_to_name(data_type), is_scalar ? SCALAR_STRIDE : COLUMN_STRIDE);
+  };
+
+  std::vector<std::string> typenames;
+
+  typenames.push_back(column_type_name(output.type(), false));
+  std::transform(
+    inputs.begin(), inputs.end(), std::back_inserter(typenames), [&](auto const& input) {
+      bool const is_scalar = input.size() != output.size();
+      return column_type_name(input.type(), is_scalar);
+    });
+
+  return typenames;
+}
+
+std::map<unsigned int, std::string> build_ptx_params(mutable_column_view output,
+                                                     std::vector<column_view> const& inputs)
```

Comment:
```suggestion
std::map<uint32_t, std::string> build_ptx_params(mutable_column_view output,
                                                     std::vector<column_view> const& inputs)
```
Use specific integer types when possible. 
Also does this need to be unsigned? The compiler and runtime require more instructions for managing unsigned types.

Reply:
I used unsigned int because it's always 32-bits and it communicates the values should be non-negative indices.
I can change it to int32_t if you feel that's a better choice.

Reply:
I see, I thought `cudf::size_type` was only meant for the size of columns

Reply:
done. thanks!